app/javascript/widget/mixins/nextAvailabilityTime.js

```diff
@@ -0,0 +1,256 @@
+import {
+  timeSlotParse,
+  defaultTimeSlot,
+} from 'dashboard/routes/dashboard/settings/inbox/helpers/businessHour.js';
+const { utcToZonedTime } = require('date-fns-tz');
+
+export default {
+  data() {
+    return {
+      dayNames: [
+        'Sunday',
+        'Monday',
+        'Tuesday',
+        'Wednesday',
+        'Thursday',
+        'Friday',
+        'Saturday',
+      ],
+      timeSlots: [...defaultTimeSlot],
+      timeSlot: {},
+    };
+  },
+  computed: {
+    channelConfig() {
+      return window.chatwootWebChannel;
+    },
+    workingHoursEnabled() {
+      const { workingHoursEnabled } = this.channelConfig;
+      return workingHoursEnabled;
+    },
+    workingHours() {
+      if (this.workingHoursEnabled) {
+        return this.channelConfig.workingHours;
+      }
+      return null;
+    },
+    timeZoneValue() {
+      if (this.workingHoursEnabled) {
+        return this.channelConfig.timezone;
+      }
+      return null;
+    },
+    currentDayWorkingHours() {
+      if (this.workingHoursEnabled) {
+        return this.workingHours.find(
+          slot => slot.day_of_week === this.currentDay
+        );
+      }
+      return null;
+    },
+    nextDayWorkingHours() {
+      if (this.workingHoursEnabled) {
+        let nextDay = this.getNextDay(this.currentDay);
+        let nextWorkingHour = this.getNextWorkingHour(nextDay);
+
+        // It gets the next working hour for the next day. If there is no working hour for the next day,
+        // it keeps iterating through the days of the week until it finds the next working hour.
+        while (!nextWorkingHour) {
+          nextDay = this.getNextDay(nextDay);
+          nextWorkingHour = this.getNextWorkingHour(nextDay);
+        }
+        return nextWorkingHour;
+      }
+      return null;
+    },
+    newDateWithTimeZone() {
+      const date = new Date();
+      const timeZone = this.timeZoneValue;
+      const zonedDate = utcToZonedTime(date, timeZone);
+      return zonedDate;
+    },
+    presentHour() {
+      return this.newDateWithTimeZone.getHours();
+    },
+    presentMinute() {
+      return this.newDateWithTimeZone.getMinutes();
+    },
+    currentDay() {
+      const date = this.newDateWithTimeZone;
+      const day = date.getDay();
+      const currentDay = Object.keys(this.dayNames).find(
+        key => this.dayNames[key] === this.dayNames[day]
+      );
+      return Number(currentDay);
+    },
+    currentDayOpenHour() {
+      return this.workingHoursEnabled
+        ? this.currentDayWorkingHours.open_hour
+        : null;
+    },
+    currentDayOpenMinute() {
+      return this.workingHoursEnabled
+        ? this.currentDayWorkingHours.open_minutes
+        : null;
+    },
+    currentDayCloseHour() {
+      return this.workingHoursEnabled
+        ? this.currentDayWorkingHours.close_hour
+        : null;
+    },
```

Comment:
This can be collapsed into one, see suggestions below. 

```suggestion
    currentDatTimings() {
      const {
        open_hour: openHour,
        open_minutes: openMinute,
        close_hour: closeHour,
      } = this.currentDayWorkingHours ?? {};
      return {
        openHour,
        openMinute,
        closeHour,
      };
    },
```

However, a better way would be to use the `currentDayWorkingHours` computed directly, and have it return sane defaults. See second code block for that change, and then we can remove the three methods alltogether

```
 currentDayWorkingHours() {
      if (!this.workingHoursEnabled) {
        return {
          openHour: null,
          openMinute: null,
          closeHour: null,
        };
      }
      const {
        open_hour: openHour,
        open_minutes: openMinute,
        close_hour: closeHour,
      } = this.workingHours.find(slot => slot.day_of_week === this.currentDay);

      return {
        openHour,
        openMinute,
        closeHour,
      };
    },
```